### PR TITLE
Allow display of unplaced labels 

### PR DESCRIPTION
**Pull request by nyalldawson**
This setting (accessed through the project label settings dialog or via an action on the labeling toolbar) allows unplaced labels to be shown on the map. These are rendered using a red color by default (but the color can be changed from the label settings dialog too), allowing users to determine whether any important labels are missing from their maps (e.g. due to overlaps or other constraints).

Additionally, there's a similar checkbox present in the labeling panel for a layout map, allowing unplaced labels in that particular layout map to be displayed.

### CODE REVIEW

**Comment by nyalldawson** [comment]
@nirvn needing your icon design skills for a proper toolbar icon here!

File: src/app/layout/qgslayoutmapwidget.cpp [hunk 1]
```diff
@@ -1601,6 +1601,7 @@ QgsLayoutMapLabelingWidget::QgsLayoutMapLabelingWidget( QgsLayoutItemMap *map )
   connect( mLabelBoundaryUnitsCombo, &QgsLayoutUnitsComboBox::changed, this, &QgsLayoutMapLabelingWidget::labelMarginUnitsChanged );
   connect( mLabelBoundarySpinBox, static_cast < void ( QDoubleSpinBox::* )( double ) > ( &QDoubleSpinBox::valueChanged ), this, &QgsLayoutMapLabelingWidget::labelMarginChanged );
   connect( mShowPartialLabelsCheckBox, &QCheckBox::toggled, this, &QgsLayoutMapLabelingWidget::showPartialsToggled );
+  connect( mDrawUnplacedCheckBox, &QCheckBox::toggled, this, &QgsLayoutMapLabelingWidget::showUnplacedToggled );
```
Comment (nirvn) [inline src/app/layout/qgslayoutmapwidget.cpp]: Cosmetic suggestion re checkbox and widget function name: settle on draw vs. show. I'd suggest sticking to show.

**Comment by nirvn** [comment]
@nyalldawson , here's an icon:
[mActionShowUnplacedLabel.zip](https://github.com/qgis/QGIS/files/3413307/mActionShowUnplacedLabel.zip)

**Comment by nyalldawson** [comment]
Thanks @nirvn, integrated now

**Comment by nirvn** [comment]
@nyalldawson , code looking good. +1 to merge when travis turns green.

**Comment by nyalldawson** [comment]
Thanks! Let qgis 3.10 forever be known as the "labelling revolution"

**Pull request merged**